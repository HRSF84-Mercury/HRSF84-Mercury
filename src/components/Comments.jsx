import React from 'react';

const Comments = ({ comments }) => {
  return (
    <div className="container column">
      {comments.map((comment) => {
        return <div key={comment._id} className="comment">{`${comment.name}: ${comment.comment}`}</div>;
      })}
    </div>
  );
};

export default Comments;

