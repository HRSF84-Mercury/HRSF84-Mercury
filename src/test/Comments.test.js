
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Comments from '../components/Comments';

describe('Comments', () => {
  it('should render correctly with props', () => {
    const comments = [{
      '_id': '_fakeId',
      'name': 'Some Awesome Name',
      'comment': 'A great comment!'
    }];
    const output = shallow(
      <Comments comments={comments} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

