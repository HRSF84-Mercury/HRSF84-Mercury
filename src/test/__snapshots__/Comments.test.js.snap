// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments should render correctly with props 1`] = `
<div
  className="container column"
>
  <div
    className="comment"
    key="_fakeId"
  >
    Some Awesome Name: A great comment!
  </div>
</div>
`;
